src/conn/conn_log.c

```diff
@@ -400,12 +401,21 @@ __log_remove_once(WT_SESSION_IMPL *session, uint32_t backup_file)
     min_lognum = backup_file == 0 ? WT_MIN(log->ckpt_lsn.l.file, log->sync_lsn.l.file) :
                                     WT_MIN(log->ckpt_lsn.l.file, backup_file);
 
+    /*
+     * Take a consistent view of the current configuration. If another thread reconfigures the
+     * connection in parallel we'll handle that change on the next call to this function.
+     */
+    __wt_spin_lock(session, &conn->reconfig_lock);
```

Comment:
Should we change this to a rwlock? We've identified 16 unique shared variables that on the connection, so our PM-3221 changes could result in multiple threads waiting to hold the `reconfig_lock`. Being able to take read locks here will reduce contention.

Prior to this PR `reconfig_lock` was only used to prevent two concurrent calls to `__wt_conn_reconfig`. I don't see a change from a spinlock to a write lock having much impact.

Reply:
I'm in favour of an rwlock.